iam-login-service/src/main/java/it/infn/mw/iam/core/oauth/IamOAuth2RequestFactory.java

```diff
@@ -164,4 +186,17 @@ public TokenRequest createTokenRequest(Map<String, String> requestParameters,
     return new TokenRequest(requestParameters, clientId, scopes, grantType);
   }
 
+  private boolean isValidUrl(String url) {
+    try {
+      new URL(url).toURI();
+      return true;
+    } catch (MalformedURLException | URISyntaxException e) {
+      throw new InvalidResourceError("Not a valid URI: " + url);
```

Comment:
In case of an exception, it's not clear to me at what level the exception is caught; in other words, does the whole request fail? just the setting of the audience? if `resource` is invalid, shall we (re)consider the `aud` parameter instead?

Reply:
can you add some tests for these cases, please?

Reply:
If the "resource" is not a valid URI, the whole request fails (see tests https://github.com/indigo-iam/iam/blob/issue-381/iam-login-service/src/test/java/it/infn/mw/iam/test/oauth/ResourceIndicatorTests.java).
At least this is my interpretation of the specification, which states

```
If the authorization server fails to parse the provided value(s) or does not consider the resource(s) acceptable, it should reject the request with an error response using the error code invalid_target as the value of the error parameter and can provide additional information regarding the reasons for the error using the error_description.
```